### PR TITLE
switch the easy admission plugins to external types

### PR DESCRIPTION
**Pull request by deads2k**
I also picked up the user change before I realized how far it rippled.

### CODE REVIEW

**Comment by deads2k** [comment]
@stevekuznetsov https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17288/test_pull_request_origin_extended_builds/629/ looks like a set up failure. Maybe something related to branching?

**Comment by stevekuznetsov** [comment]
@deads2k that should be fixed now 

/retest

**Comment by deads2k** [comment]
Alright, I've convinced myself that it isn't going to be easier to break our internal api dependency on internal kube apis.

File: pkg/build/admission/strategyrestrictions/admission.go [hunk 1]
```diff
@@ -206,5 +215,19 @@ func (a *buildByStrategy) checkAccess(strategy buildapi.BuildStrategy, subjectAc
 }
 
 func notAllowed(strategy buildapi.BuildStrategy, attr admission.Attributes) error {
-	return admission.NewForbidden(attr, fmt.Errorf("build strategy %s is not allowed", buildapi.StrategyType(strategy)))
+	return admission.NewForbidden(attr, fmt.Errorf("build strategy %s is not allowed", strategyTypeString(strategy)))
+}
+
+func strategyTypeString(strategy buildapi.BuildStrategy) string {
```
Comment (mfojtik) [inline pkg/build/admission/strategyrestrictions/admission.go]: seems like build api helper to me

File: pkg/build/admission/strategyrestrictions/admission_test.go [hunk 2]
```diff
@@ -250,6 +254,15 @@ func testBuild(strategy buildapi.BuildStrategy) *buildapi.Build {
...
 }
 
+func asV1Build(in *buildapi.Build) *buildapiv1.Build {
+	out := &buildapiv1.Build{}
+	err := kapi.Scheme.Convert(in, out, nil)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
```
Comment (bparees) [inline pkg/build/admission/strategyrestrictions/admission_test.go]: also seems like a helper.

File: pkg/build/admission/strategyrestrictions/admission_test.go [hunk 3]
```diff
@@ -264,6 +277,15 @@ func testBuildConfig(strategy buildapi.BuildStrategy) *buildapi.BuildConfig {
...
 }
 
+func asV1BuildConfig(in *buildapi.BuildConfig) *buildapiv1.BuildConfig {
+	out := &buildapiv1.BuildConfig{}
+	err := kapi.Scheme.Convert(in, out, nil)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
```
Comment (bparees) [inline pkg/build/admission/strategyrestrictions/admission_test.go]: as does this

File: pkg/build/admission/strategyrestrictions/admission.go (hunk 1 above)
Comment (deads2k) [reply pkg/build/admission/strategyrestrictions/admission.go]: > seems like build api helper to me

Single point of use is here.

File: pkg/build/admission/strategyrestrictions/admission_test.go (hunk 2 above)
Comment (deads2k) [reply pkg/build/admission/strategyrestrictions/admission_test.go]: Single point of use is here.

File: pkg/build/admission/strategyrestrictions/admission_test.go (hunk 3 above)
Comment (deads2k) [reply pkg/build/admission/strategyrestrictions/admission_test.go]: Single point of use is here.

File: pkg/build/admission/strategyrestrictions/admission_test.go (hunk 2 above)
Comment (bparees) [reply pkg/build/admission/strategyrestrictions/admission_test.go]: you realize i'm probably about to have to do this everywhere else as part of converting to v1 apis, right?

**Pull request merged**